chemprop/data/data.py

```diff
@@ -179,7 +179,11 @@ def __init__(self,
     @property
     def mol(self) -> List[Union[Chem.Mol, Tuple[Chem.Mol, Chem.Mol]]]:
         """Gets the corresponding list of RDKit molecules for the corresponding SMILES list."""
-        mol = make_mols(self.smiles, self.is_reaction_list, self.is_explicit_h_list, self.is_adding_hs_list, self.is_keeping_atom_map_list)
+        if self.atom_targets is not None or self.bond_targets is not None:
+            # When the original atom mapping is used, the explicit hydrogens specified in the input SMILES are used
+            mol = make_mols(self.smiles, self.is_reaction_list, self.is_keeping_atom_map_list, self.is_adding_hs_list, self.is_keeping_atom_map_list)
```

Comment:
Based on the definition of "make_mols"

def make_mols(smiles: List[str], reaction_list: List[bool], keep_h_list: List[bool], add_h_list: List[bool], keep_atom_map_list: List[bool]):

I think it might be worth to rename or add comments to the "self.is_keeping_atom_map_list" argument that you are passing to "make_mols". Right not "self.is_keeping_atom_map_list" is repeated twice. Although I think you intended to do so, but this is a bit confusing to read. Another minor point: "keep_h_list" reads like it is a list to specify only about which Hs to be kept (not include heavy atoms), and "keep_atom_map_list" seems to be more general (include H and heavy atoms). Therefore, passing the same input to specify these two arguments is just confusing. Maybe also considering further explain the difference in the docstring, which is not being done in "make_mols".

Reply:
The arguments (self.is_explicit_h_list, self.is_adding_hs_list, self.is_keeping_atom_map_list) passed into `make_mols` are lists of booleans to determine whether to keep hydrogens, add hydrogens, or to keep the original atom mapping. This is already documented in the `make_mols` function. The reason for assigning `is_keeping_atom_map_list` as `is_keeping_atom_map_list` is because keeping explicit Hs can only be used for reaction SMILES. However, we expect that the explicit Hs should be added when `--keeping_atom_map` flag is used. Do you think adding a comment related to the above explanation is enough?